tabpfn_client/estimator.py

```diff
@@ -393,3 +409,23 @@ def _model_name_to_path(self, model_name: str) -> str:
             return f"{base_path}_{model_name}.ckpt"
         else:
             raise ValueError(f"Invalid model name: {model_name}")
+
+
+def validate_data_size(X: np.ndarray, y: np.ndarray | None = None):
+    """
+    Check the integrity of the training data.
+    - check if the number of rows between X and y is consistent
+        if y is not None (ValueError)
+    - check if the number of rows is less than MAX_ROWS (ValueError)
+    - check if the number of columns is less than MAX_COLS (ValueError)
+    """
+
+    # check if the number of samples is consistent (ValueError)
+    if y is not None:
+        check_consistent_length(X, y)
```

Comment:
since the rest of the code relies on `.shape` to exist, can you simply use that here too s.t. it is directly understandable what you are doing

Reply:
Alright, done!

Reply:
thanks